tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Logging/Log4Net/AppenderAttachedImplIntegration.cs

```diff
@@ -0,0 +1,74 @@
+// <copyright file="AppenderAttachedImplIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.ComponentModel;
+using Datadog.Trace.ClrProfiler.CallTarget;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Log4Net
+{
+    /// <summary>
+    /// LoggerFactoryScopeProvider.ForEach&lt;TState&gt; calltarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = "log4net",
+        TypeName = "log4net.Util.AppenderAttachedImpl",
+        MethodName = "AppendLoopOnAppenders",
+        ReturnTypeName = ClrNames.Int32,
+        ParameterTypeNames = new[] { "log4net.Core.LoggingEvent" },
+        MinimumVersion = "1.0.0",
+        MaximumVersion = "2.*.*",
+        IntegrationName = "Log4Net")]
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class AppenderAttachedImplIntegration
+    {
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TLoggingEvent">The type of the logging event</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="loggingEvent">The logging event</param>
+        /// <returns>Calltarget state value</returns>
+        public static CallTargetState OnMethodBegin<TTarget, TLoggingEvent>(TTarget instance, TLoggingEvent loggingEvent)
```

Comment:
Feel free to ignore it for now, but in my "direct log submission" work, I moved the logs-injection related integrations into a _LogsInjection_ subfolder, e.g. _Logging/Log4Net/LogsInjection_. Don't know if that's something you want to do now, or I can do it later?

Reply:
I'd like to keep the current organization in this PR and in your PR do the _LogsInjection_ migration. Is that cool?